core/src/main/scala/org/apache/spark/deploy/PythonRunner.scala

```diff
@@ -92,15 +92,19 @@ object PythonRunner {
       // see https://github.com/numpy/numpy/issues/10455
       sparkConf.getOption("spark.driver.cores").foreach(env.put("OMP_NUM_THREADS", _))
     }
-    builder.redirectErrorStream(true) // Ugly but needed for stdout and stderr to synchronize
     try {
       val process = builder.start()
 
+      val stderrBuffer = new CircularBuffer(2048)
+      new RedirectThread(
+        process.getErrorStream,
+        stderrBuffer,
+        "Thread-ScriptTransformation-STDERR-Consumer").start()
       new RedirectThread(process.getInputStream, System.out, "redirect output").start()
```

Comment:
Looks like now it won't show the stderr in stdout anymore?

Reply:
> Looks like now it won't show the stderr in stdout anymore?

It's a problem.   If we reserve `builder.redirectErrorStream(true)`
We need to split message in inputStream and got error message after  `Traceback` as error message?

Reply:
Is there any easy way to get the stderr stream, and manually merge stderr and stdout to printout? I think we shouldn't remove this from stdout.

Reply:
> Is there any easy way to get the stderr stream, and manually merge stderr and stdout to printout? I think we shouldn't remove this from stdout.

How about current?